### PR TITLE
[WIP] YarpFixManager: fix after recent changes in libYARP_manager

### PR DESCRIPTION
**Pull request by Nicogene**
This PR introduces the changes for making the `YarpFixManager` compatible after recent changes in `libYARP_manager`.
They are the same of robotology/yarp#1242
MERGE AFTER robotology/yarp#1244
Please review code.

### CODE REVIEW

File: src/middleware/yarp/fixture-ymanager/YarpFixManager.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ bool YarpFixManager::setup(int argc, char** argv) {
         enableRestrictedMode();
 
         // load the fixture (application xml)
-        char szAppName[255];
+        char* szAppName = new char [10];
```
Comment (drdanz) [inline src/middleware/yarp/fixture-ymanager/YarpFixManager.cpp]: Why 10?

Comment (drdanz) [inline src/middleware/yarp/fixture-ymanager/YarpFixManager.cpp]: Why allocating it here then?

Comment (traversaro) [inline src/middleware/yarp/fixture-ymanager/YarpFixManager.cpp]: I don't understand. The actual (pointer) value of `szAppName` seems to be ignored by `KnowledgeBase::addApplication`, and given that it is a pointer passed by value, it is not useful as an output parameter either. Why it is a argument of the `KnowledgeBase::addApplication` method at all?

Comment (drdanz) [inline src/middleware/yarp/fixture-ymanager/YarpFixManager.cpp]: You are the expert! :stuck_out_tongue_closed_eyes: 
I'm just worried that these 10 bytes are leaked somewhere...

**Pull request closed without merging**